charts/orchestrator/templates/_helpers.tpl

```diff
@@ -120,4 +120,21 @@
         {{- end -}}
         {{- .argoCDNamespace -}}
     {{- end -}}
+{{- end -}}
+
+{{- define "get-cluster-version" -}}
+  {{- $v := "" }}
+  {{- $version :=(lookup "config.openshift.io/v1" "ClusterVersion" "" "version") }}
+  {{- range $version.status.history }}
+    {{- if eq .state "Completed" }}
+      {{- $v = (semver .version) }}
+    {{- end }}
+  {{- end }}
+
+  {{- $validMinors := list "13" "14" "15" "16" -}}
+  {{- $versionString := printf "%d.%d" $v.Major $v.Minor -}}
+  {{- if not (semverCompare ">=4.13 <=4.16" $versionString) -}}
+    {{- fail (printf "Unsupported OCP version: %s. Supported versions: %s." $versionString $validMinors) -}}
```

Comment:
Shouldn't it be printing the list contents with `%v` perhaps?

Reply:
I'll change to %v, but the output is the same:
```
with %s
→ helm upgrade --install orchestrator . -n orchestrator -f ./values-rc.yaml --dry-run=server
Error: UPGRADE FAILED: execution error at (orchestrator/templates/release-candidate.yaml:9:17): Unsupported OCP version: 5.14. Supported versions: [4.13 4.14 4.15 4.16].
```

```
with %v
→ helm upgrade --install orchestrator . -n orchestrator -f ./values-rc.yaml --dry-run=server
Error: UPGRADE FAILED: execution error at (orchestrator/templates/release-candidate.yaml:9:17): Unsupported OCP version: 5.14. Supported versions: [4.13 4.14 4.15 4.16].
```

Reply:
`%s` it is then.